lib/y2_module_consoletest.pm

```diff
@@ -0,0 +1,55 @@
+# Base for YaST tests.  Switches to text console 2 and uploady y2logs
+
+package y2_module_consoletest;
+use parent 'y2_module_basetest';
+use y2_installbase qw(save_upload_y2logs save_system_logs save_strace_gdb_output);
+use strict;
+use warnings;
+use testapi;
+use utils 'show_tasks_in_blocked_state';
+
+sub yast2_console_exec {
+    my %args = @_;
+    die "Yast2 module has not been found among function arguments!\n" unless (defined($args{yast2_module}));
+    my $y2_start    = 'Y2DEBUG=1 ZYPP_MEDIA_CURL_DEBUG=1 yast2 ';
+    my $module_name = 'yast2-' . $args{yast2_module} . '-status';
+    $y2_start .= (defined($args{yast2_opts})) ?
+      $args{yast2_opts} . ' ' . $args{yast2_module} . ';' :
+      $args{yast2_module} . ';';
+
+    if (!script_run($y2_start . " echo $module_name-\$? > /dev/$serialdev", 0)) {
+        return $module_name;
+    } else {
+        die "Yast2 module failed to execute!\n";
+    }
+}
+
+sub post_fail_hook {
+    my $self = shift;
+
+    my $defer_blocked_task_info = testapi::is_serial_terminal();
+    show_tasks_in_blocked_state unless ($defer_blocked_task_info);
+
+    select_console 'log-console';
+    save_screenshot;
+
+    show_tasks_in_blocked_state if ($defer_blocked_task_info);
+
+    $self->remount_tmp_if_ro;
+    save_upload_y2logs;
+    upload_logs('/var/log/zypper.log', failok => 1);
+    save_system_logs;
+    save_strace_gdb_output('yast');
```

Comment:
The functions called without $self:
- save_upload_y2logs
- save_system_logs
- save_strace_gdb_output

It will cause an error. Please, search for:
```post_fail_hook failed: Can't call method "investigate_yast2_failure" on an undefined value at /var/lib/openqa/pool/10/os-autoinst-distri-opensuse/lib/y2_installbase.pm line 190.``` 

to see the example in the verification run logs: https://openqa.suse.de/tests/2886984/file/autoinst-log.txt

Reply:
fixed